geo/src/algorithm/rotate.rs

```diff
@@ -1,44 +1,22 @@
-use crate::AffineTransform;
-use crate::BoundingRect;
-use crate::Centroid;
-use crate::MapCoords;
-use crate::MapCoordsInPlace;
-use crate::{
-    CoordFloat, Coordinate, GeoFloat, Line, LineString, MultiLineString, MultiPoint, MultiPolygon,
-    Point, Polygon,
-};
-
-// Rotate an iterator of points "angle" degrees about an origin. Origin can be
-// an arbitrary point. Pass Point::new(0., 0.) for the actual origin.
-fn rotate_many<'a, T>(
-    angle: T,
-    origin: Point<T>,
-    points: impl Iterator<Item = &'a Coordinate<T>> + 'a,
-) -> impl Iterator<Item = Coordinate<T>> + 'a
-where
-    T: CoordFloat,
-    T: 'a,
-{
-    let affineop = AffineTransform::rotate(angle, origin);
-    points.map(move |coord| affineop.apply(*coord))
-}
-
-pub trait Rotate<T> {
+use crate::algorithm::{AffineOps, AffineTransform, BoundingRect, Centroid};
+use crate::geometry::*;
+use crate::CoordFloat;
+
+/// Rotate a geometry around a point by an angle, in degrees.
+///
+/// Positive angles are counter-clockwise, and negative angles are clockwise rotations.
+///
+/// ## Performance
+///
+/// If you will be performing multiple transformations, like [`Scale`](crate::Scale),
+/// [`Skew`](crate::Skew), [`Translate`](crate::Translate), or [`Rotate`](crate::Rotate), it is more
+/// efficient to compose the transformations and apply them as a single operation using the
+/// [`AffineOps`](crate::AffineOps) trait.
+pub trait Rotate<T: CoordFloat> {
     /// Rotate a geometry around its [centroid](Centroid) by an angle, in degrees
     ///
     /// Positive angles are counter-clockwise, and negative angles are clockwise rotations.
     ///
-    /// # Units
-    ///
-    /// - `angle`: degrees
```

Comment:
After renaming the parameter to `degrees` this documentation felt pretty superfluous.